lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -202,36 +232,13 @@ export default function getRecipeInterface(
                     return undefined;
                 }
 
-                logDebugMessage("getSession: UNAUTHORISED because accessToken from cookies is undefined");
+                logDebugMessage("getSession: UNAUTHORISED because accessToken in request is undefined");
                 throw new STError({
                     message: "Session does not exist. Are you sending the session tokens in the request as cookies?",
                     type: STError.UNAUTHORISED,
                 });
             }
-            if (accessToken === undefined) {
-                // maybe the access token has expired.
-                /**
-                 * Based on issue: #156 (spertokens-node)
-                 * we throw TRY_REFRESH_TOKEN only if
-                 * options.sessionRequired === true || (frontendHasInterceptor or request method is get),
-                 * else we should return undefined
-                 */
-                if (
-                    options === undefined ||
-                    (options !== undefined && options.sessionRequired === true) ||
-                    frontendHasInterceptor(req) ||
-                    normaliseHttpMethod(req.getMethod()) === "get"
-                ) {
-                    logDebugMessage(
-                        "getSession: Returning try refresh token because access token from cookies is undefined"
-                    );
-                    throw new STError({
-                        message: "Access token has expired. Please call the refresh API",
-                        type: STError.TRY_REFRESH_TOKEN,
-                    });
-                }
-                return undefined;
-            }
+
             try {
                 let antiCsrfToken = getAntiCsrfTokenFromHeaders(req);
                 const disableAntiCSRF = transferMethod === "header";
```

Comment:
i know what `disableAntiCSRF` and `doAntiCsrfCheck` mean in the context of this code and that you are doing `!disableAntiCSRF && doAntiCsrfCheck,` below. 

But as a new reader of this code, it can be very confusing as to why you have two booleans that mean the same thing (semantically), but can have opposing values at the same time.